app/src/main/java/com/github/gotify/sharing/ShareActivity.java

```diff
@@ -0,0 +1,174 @@
+
+                        EditText edtTxtContent = findViewById(R.id.content);
+                        String content = edtTxtContent.getText().toString();
+                        message.setMessage(content);
+
+                        Spinner appSpinner = findViewById(R.id.appSpinner);
+                        String selectedApp = appSpinner.getSelectedItem().toString();
+
+                        Spinner prioritySpinner = findViewById(R.id.prioritySpinner);
+                        Long priority =
+                                Long.parseLong(prioritySpinner.getSelectedItem().toString());
+                        message.setPriority(priority);
+
+                        new GetApps(selectedApp).execute(message);
+                    }
+                });
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        if (item.getItemId() == android.R.id.home) {
+            finish();
+        }
+        return super.onOptionsItemSelected(item);
+    }
+
+    private void PopulateSpinner(List<Application> apps) {
+        List<String> appNameList = new ArrayList();
+        for (Application app : apps) {
+            appNameList.add(app.getName());
+        }
+        Spinner dynamicSpinner = findViewById(R.id.appSpinner);
+        ArrayAdapter<String> adapter =
+                new ArrayAdapter<String>(
+                        ShareActivity.this,
+                        android.R.layout.simple_spinner_dropdown_item,
+                        appNameList);
+        dynamicSpinner.setAdapter(adapter);
+        dynamicSpinner.setOnItemSelectedListener(
+                new AdapterView.OnItemSelectedListener() {
+                    @Override
+                    public void onItemSelected(
+                            AdapterView<?> parent, View view, int position, long id) {
+                        // Log.v("item", (String) parent.getItemAtPosition(position));
+                    }
+
+                    @Override
+                    public void onNothingSelected(AdapterView<?> parent) {
+                        // TODO Auto-generated method stub
+                    }
+                });
+
+        // Auto select app
+        Integer selectedApp = getIntent().getIntExtra("SELECTED_APP", 0);
+        dynamicSpinner.setSelection(selectedApp);
+    }
+
+    private class GetApps extends AsyncTask<Message, Void, Void> {
+        private String SelectedApp;
+
+        public GetApps(String passedData) {
+            SelectedApp = passedData;
+        }
+
+        @SuppressLint("WrongThread")
+        @Override
+        protected Void doInBackground(Message... messages) {
+            try {
+                ApplicationApi applicationApi = client.createService(ApplicationApi.class);
+                List<Application> apps = Api.execute(applicationApi.getApps());
+                for (Application app : apps) {
+                    if (app.getName().equals(SelectedApp)) {
+                        client =
```

Comment:
Don't reassign the client, instead pass the Application and Message to the AsyncTask and then create a client just for the request.